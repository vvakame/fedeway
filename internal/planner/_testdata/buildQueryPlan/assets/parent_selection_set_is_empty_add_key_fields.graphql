# schema: supergraphSdl.graphqls

# when the parent selection set is empty
# should add key fields to the parent selection set and use a dependent fetch
query {
    topReviews {
        author {
            name {
                first
            }
        }
    }
}
