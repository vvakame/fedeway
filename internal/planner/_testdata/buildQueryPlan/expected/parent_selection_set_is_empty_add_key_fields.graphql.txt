QueryPlan {
	Sequence {
		Fetch(service: "reviews") {
			query {
				topReviews {
					author {
						__typename
						id
					}
				}
			}
		},
		Flatten(path: "topReviews.@.author") {
			Fetch(service: "accounts") {
				{
					... on User {
						__typename
						id
					}
				} =>
				{
					... on User {
						name {
							first
						}
					}
				}
			},
		},
	},
}